api/v1alpha1/envoygateway_types.go

```diff
@@ -480,10 +480,18 @@ type XDSTranslatorHooks struct {
 	Post []XDSTranslatorHook `json:"post,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="(has(self.Path) && !has(self.Host)) || (has(self.Host) && !has(self.Path))",message="transport configuration must be either tcp or uds"
 // ExtensionService defines the configuration for connecting to a registered extension service.
 type ExtensionService struct {
+	// Path defines the path to the extension service Unix Domain Socket file.
+	//
+	// +optional
+	Path string `json:"path,omitempty"`
+
 	// Host define the extension service hostname.
-	Host string `json:"host"`
+	//
+	// +optional
```

Comment:
I'd prefer if we deprecated the current fields and added something that looked like
https://github.com/envoyproxy/gateway/blob/6291a370d5ba965607f1dd2121793ca2e88f9f0a/api/v1alpha1/backend_types.go#L58 e.g.
```
type: Unix
unix:
  path: /path/to/domain.sock
```

```
type: FQDN
fqdn:
  hostname: name.ns.svc.cluster.local
  port: 8080
```

Reply:
Done